### DISCUSSION TITLE
Take some ideas from tantalum

### DISCUSSION THREAD

**Discussion by vshymanskyy** [ricktu288/ray-optics#138] [Ideas]
FYI -> https://benedikt-bitterli.me/tantalum/
![image](https://user-images.githubusercontent.com/1506708/43087331-443023ec-8ea8-11e8-89d8-8348461d4a70.png)

**Comment by j3soon** [comment]
Wow! The project is so beautiful and well documented! Our project has also encountered some of the problem mentioned in its documentation (e.g. the rasterization bias problem, etc).
However, the project serves a different purpose than ours. We wish to let users to drag around different objects and observe the simulated result.

**Comment by j3soon** [comment]
I'll keep this issue open for now to let more people see this awesome project! Thanks @vshymanskyy !

**Comment by vshymanskyy** [comment]
@j3soon Thanks for a quick response. I'm particularly interested in spectral rendering feature the tantalum implemented. It's cool to have an editor, and your project already has. You can borrow some ideas for the simulator itself, I mean.
Anyway, just wanted to share my findings!

**Comment by j3soon** [comment]
@vshymanskyy We'll think about it. Tantalum uses WebGL to optimize rendering, however, our project uses HTML 5 canvas to achieve good compatibility. So it might not be easy to implement those rendering features using canvas.
By the way, there's a app called [RayLab](http://www.raymak.com/wp/), which also has many cool features.

**Comment by ricktu288** [comment]
GPU ray tracing is now implemented and can be turned on in Settings -> Engine -> WebGPU
